crates/subspace-networking/src/create.rs

```diff
@@ -197,6 +220,7 @@ pub async fn create(config: Config) -> Result<(Node, NodeRunner), CreationError>
 
                 Ok((peer_id, multiaddr))
             })
+            .chain(cached_bootstrap_addresses.into_iter().map(Ok))
```

Comment:
`cached_bootstrap_addresses` should be the primary source with `bootstrap_nodes` as a fallback in case no or very few peers are found. 

Something like this:
```rust
cached_bootstrap_addresses
    .into_iter()
    .map(Ok)
    .chain(
        bootstrap_nodes
            .into_iter()
            .map(|mut multiaddr| {
                let peer_id: PeerId = multiaddr
                    .pop()
                    .and_then(|protocol| {
                        if let Protocol::P2p(peer_id) = protocol {
                            Some(peer_id.try_into().ok()?)
                        } else {
                            None
                        }
                    })
                    .ok_or(CreationError::BadBootstrapAddress)?;

                Ok((peer_id, multiaddr))
            })
    )
    .take(INITIAL_BOOTSTRAP_ADDRESS_NUMBER)
    .collect::<Result<_, CreationError>>()?;

Reply:
> Manually set parameters seem to be more important than autogenerated and hidden ones. What if a user want to add new addresses to an already populated cache?

Typically bootstrap nodes (regardless of whether they are defaults or specified explicitly) are lower priority. If we want to have special nodes to which connection needs to be maintained, we need to replicate something similar to reserved peers in Substrate instead. Otherwise there isn't much point in new code, bootstrap nodes are still reach on each start just like before.